forum/qa-plugin/outdated-question-info/frontend/show-outdated-question-info.js

```diff
@@ -6,29 +6,30 @@ if(placeOfOutdatedQuestionInfo){
     const publishYearOlderThanNow = publishDate.getFullYear() < now.getFullYear();
     const publishMonthNewerThanNow = publishDate.getMonth() - 1 >= now.getMonth();
 
-    if (publishYearOlderThanNow && publishMonthNewerThanNow) {
-        let shouldBeDisplayed = "";
-        if(placeOfOutdatedQuestionInfo.style.display === 'none'){
-            shouldBeDisplayed = "hidden";
-        }
+    if (publishYearOlderThanNow || publishMonthNewerThanNow) {
+        const outdatedQuestionInfoVisibility = placeOfOutdatedQuestionInfo.style.display === 'none' ? 'hidden' : '';
         placeOfOutdatedQuestionInfo.insertAdjacentHTML('beforebegin', 
-                `<p class = "qa-outdated-question-container ${shouldBeDisplayed}">
+                `<p class = "qa-outdated-question-container ${outdatedQuestionInfoVisibility}">
                     To pytanie zostało zadane już dawno temu i może być nieaktualne.<br/>
                     Upewnij się, że Twoja odpowiedź nadal będzie pomocna.
                 </p>`);
     }
-    const QuestionElemExist = document.querySelector('.qa-outdated-question-container');
+    const questionElemExist = document.querySelector('.qa-outdated-question-container');
+    if(questionElemExist){
+        const outdatedInfoContainerClassList = questionElemExist.classList;
+    
+        const areThereAnswersForQuestion = !CKEDITOR.instances.a_content ? true : false;
+        
 
-    if(QuestionElemExist){
-        const cancelAnswer = document.querySelectorAll('input[name=docancel]')[1];
-        const outdatedInfoContainerClassList = QuestionElemExist.classList;
-
-        cancelAnswer.addEventListener('click', ()=>{
-            outdatedInfoContainerClassList.toggle('hidden');
-        }, false)
 
         document.querySelector('#q_doanswer').addEventListener('click', ()=>{
-            outdatedInfoContainerClassList.toggle('hidden');
+            outdatedInfoContainerClassList.contains('hidden') ? outdatedInfoContainerClassList.remove('hidden') : outdatedInfoContainerClassList.add('hidden');
```

Comment:
Dlaczego tutaj `.toggle()` zostało zamienione na ręczne sprawdzanie obecności klasy i odpowiednio jej usunięcie lub dodanie? Przecież do tego właśnie służy `.toggle()`. :)

Reply:
To w końcu co znaczy "eksplicytnie dodawać". Wspominałeś o tym, i tak to zinterpretowałem. Czyli wcześniejsze tooglowanie było dobrze?

Reply:
Ah, zapomniałem, że o tym pisałem wyżej. Chodziło mi o to, żeby na kliknięcie w `cancelAnswer` dodać klasę 'hidden', a na kliknięcie w '#q_doanswer' ją usunąć - wtedy eksplicytnie określa się, że anulowanie odpowiedzi ukrywa komunikat, a kliknięcie w dodanie odpowiedzi, pokazuje komunikat. Togglowanie mogło by przez pomyłkę zrobić odwrotnie i jest troszkę mniej czytelnie, jeśli się spojrzy na to od strony intencji.

Czyli po prostu:
```js
cancelAnswer.addEventListener('click', () => outdatedInfoContainerClassList.add('hidden'));

document.querySelector('#q_doanswer').addEventListener('click', () => {
            outdatedInfoContainerClassList.remove('hidden');

           // reszta kodu
});
```

Jeszcze tak ogólnie: sprawdziłbym, czy listenery będą nadal działać, jeśli:
1. klikniesz w dodanie odpowiedzi
2. klikniesz w anuluj
3. znowu klikniesz w dodanie odpowiedzi

, bo tam korzystasz z opcji `{ once: true }`, więc listener wtedy zadziała tylko raz.

Reply:
Ok, dzięki za rozjaśnienie

A co do tych listenerów, wszystko u mnie działała jak powinno. `once: true` jest tam po to, aby listener dodał się raz, nie więcej. Bez tego listenery dodawały się za każdym razem, gdy kliknąłem przycisk doanswer kod wywoływał się więcej niż 1 raz. W tej wersji wszystko działa poprawnie